2024_NewBeginnings/pr05_2024_Final_report_DoraDo.md

```diff
@@ -0,0 +1,49 @@
+  
+- App Build and Distribution: 
+  - We initially tested app distribution using electron-builder, but challenges in handling Apple’s certificate verification led to adopting electron-forge, which streamlined the macOS notarization process. electron-forge provided more reliable handling for platform-specific requirements, making it easier to sign and notarize the app for Mac and set up for Windows. There was also an issue with notarizing the Processing library, which was resolved by creating a script called `sign_jnilib_files.js` to sign the library for notarization.
+  - In order for others to build the app, contributors can use the provided `.env.example` file to set up their environment variables. 
+
+
+## Tasks Completed
+- Created a functional prototype of a collaborative code editor for Processing using Electron and CodeMirror.
+- Allow creation, editing, renaming, deleting, and running of local Processing sketches.
+- Integrated YJS with CodeMirror for real-time collaboration, enabling multiple users to edit a Processing sketch simultaneously.
+- Allows user selection of dark or light mode for the editor.
+- Builds for macOS and Windows.
+
+
+## Challenges
+- Certificate and Notarization Issues: Encountering "damaged app" errors when testing macOS builds on other machines highlighted the need for proper app signing. After troubleshooting with Apple Developer tools and certificates, switching to electron-forge resolved this issue.
+
+- Cross-Platform Compatibility: Distribution on Windows and Linux introduced different runtime and dependency challenges. Upcoming tasks will focus on thoroughly testing builds on these platforms.
+
+- UI Bug Fixes: Addressing minor yet crucial UI bugs required repeated testing. Adding collapsible panels and submenus also introduced layout adjustments that required careful tuning to avoid disrupting the user flow.
+
+
+## Limitations
+- Apple Developer Account Dependency: Open-source contributors must have Apple Developer credentials for macOS builds, which may limit participation without a centralized solution.
+- Websocket server: A crude websocket server was used for real-time collaboration, which may not be scalable for larger groups. Future iterations will explore more robust solutions for handling collaborative editing.
+- Processing Java library and JDK storage: For this project, the Processing Java library, nor the JDKm were committed to the repository due to its size. Contributors must download the Processing4 application, go into it's Contents folder and copy the Java folder into the tools/PlugIns folder. This is a temporary solution until a more automated process is implemented.
+- Windows build: The Windows build has not been fully developed or tested, since my development environment is macOS. Future work will focus on ensuring the app runs smoothly on Windows systems.
+
```

Comment:
Please add note on single tab experience: We didn't dig in how to handle multi-file (multi-tab) collaboration, as each room assumes a single file at the moment.